### PR TITLE
Add a Numba power-on-self-test script and use in CI.

### PR DESCRIPTION
**Pull request by stuartarchibald**
This replaces the much broader "test of runtests" check performed in CI with a more simple check aiming to ensure that "runtests" works and Numba can correctly compile and execute at least one thing. The intent of this change is to drastically reduce the run time spent on this check.

### CODE REVIEW

File: numba/misc/POST.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        raise AssertionError(msg)
+
+
+def check():
+    _check_runtests()
+    _check_cpu_compilation()
+
+
+if __name__ == "__main__":
+    check()
```
Comment (stuartarchibald) [inline numba/misc/POST.py:34]: Maybe this should print a status?

**Comment by stuartarchibald** [comment]
For run time comparison, it's probably best to look at the Windows builds as they have an isolated "Verify runtests" step on `main` which is the comparable step to the `Power-On-Self-Test...` step in this branch. The windows build machines are at present always skylake-avx512 chips with 4 cores available, so CPU differences are not present as they are in e.g. the linux build machines. Results:
* `main` (at 4f16b8f) build in Azure CI has step "Verify runtests" taking 2m 16s.
* this branch (at da12f2c) build in Azure CI has step "Power-On-Self-Test..." taking 9s.

Essentially, this patch makes it so that the actual testing of a change for e.g. a PR build will now start 2 minutes earlier (on windows at least)!

**Comment by stuartarchibald** [comment]
Marking this for review so as to get some feedback.

File: buildscripts/incremental/test.cmd [hunk 2]
```diff
@@ -17,8 +17,10 @@ set PYTHONFAULTHANDLER=1
 @rem enable new style error handling
 set NUMBA_CAPTURED_ERRORS=new_style
 
-@rem First check that the test discovery works
-python -m numba.tests.test_runtests
+@rem First run Numba's Power-On-Self-Test to make sure testing will likely work
```
Comment (sklam) [inline buildscripts/incremental/test.cmd:20]: Is this file used?

File: numba/misc/POST.py (hunk 1 above)
Comment (sklam) [reply numba/misc/POST.py:34]: nah, the traceback is probably good enough

File: buildscripts/incremental/test.cmd (hunk 2 above)
Comment (sklam) [reply buildscripts/incremental/test.cmd:20]: The `test.cmd` doesn’t seem to be referenced anywhere. But, i’ll ignore it for the context of this PR and open a different PR to try to remove it.

**Comment by stuartarchibald** [comment]
This is continued in #9191, it removes the changes to `test.cmd` so that it can be merged alongside #9155, squashes the patch series and rebases on `main` to accommodate changes in testing due to the RVSDG work.

**Pull request closed without merging**